tests/Model/Customer/CustomerListTest.php

```diff
@@ -554,8 +554,139 @@ public function testSingleSearchQueryFilter() {
         $modifiedListing = $handler->getListing();
         $this->assertEquals(5, $modifiedListing->getCount());
 
-    }
+        // -- Tests for Exact Search (Field should must match exactly)
+
+        // ------------
+        $listing = new Customer\Listing();
+        $handler = new FilterHandler($listing);
+
+        $searchFilter = new SearchQuery(['firstname'], '"jane"');
+        $handler->addFilter ($searchFilter);
+
+        $modifiedListing = $handler->getListing ();
+        $this->assertEquals (1, $modifiedListing->getCount ());
+
+        $listing = new Customer\Listing();
+        $handler = new FilterHandler($listing);
+
+        $searchFilter = new SearchQuery(['firstname', 'lastname'], '"jane"');
+        $handler->addFilter ($searchFilter);
+
+        $modifiedListing = $handler->getListing ();
+        $this->assertEquals (1, $modifiedListing->getCount ());
```

Comment:
weird, doesn't this should returning and expect `0` as there is no Customer with both firstname and lastname as "jane"?

Potentially the tests by passing multiple columns in the constructor's first parameter are somehow wrong.
https://github.com/pimcore/customer-data-framework/blob/f028d4ce886814b9c1f8418c235fc08f710da38c/src/CustomerList/SearchHelper.php#L118-L124
I see that it loops for every `$databaseFields` (firstname,lastname,customer id, state,gender etc..) that is filled with a value.

Reply:
![image](https://user-images.githubusercontent.com/97134765/187686243-6ea9588a-a777-4c06-a2c1-b2ce9fae44c7.png)

When we search in this input filter, it should loop through every fileds like (firstname,lastname,customer id, state,gender etc..), right. Since we can search without specifying a field name.

Reply:
ah i see but, but can we then do the tests with these fields?
![image](https://user-images.githubusercontent.com/6014195/187687323-be11684e-c644-43cc-8819-002c56c39783.png)
as the problem faced in https://github.com/pimcore/customer-data-framework/pull/317 were related to this area

Reply:
https://github.com/pimcore/customer-data-framework/pull/322/files#:~:text=%24searchFilter%20%3D%20new%20SearchQuery(%5B%27firstname%27%5D%2C%20%27%22jan%22%27)%3B - This corresponds to a kind of test case you mentioned since the number of fields is 1 and the query string ("jane" doesn't contain  "AND", "OR", "!" and "*". If required, I can test cases for all the fields like we have now in "firstname"

Reply:
ahh ok, now i think i got it, thank you!
no need for the other fields, should all behaving the same

Reply:
Yes, It depends on the number of fields and the query like ("AND", "OR", "!") etc. So, with these test cases, it should test for both the cases.